src/providers/twitch/TwitchChannel.hpp

```diff
@@ -218,8 +227,7 @@ class TwitchChannel final : public Channel, public ChannelChatters
     pajlada::Signals::NoArgSignal roomModesChanged;
 
     // Channel point rewards
-    pajlada::Signals::SelfDisconnectingSignal<ChannelPointReward>
-        channelPointRewardAdded;
+    boost::circular_buffer_space_optimized<QueuedRedemption> waitingRedemptions;
```

Comment:
This has the advantage of automatically dropping the oldest queued redemptions when many new redemptions are being queued

Imagine a redemption (for an unknown reward) occurs while pubsub is disconnected. Later, the reward is deleted and pubsub reconnects. With a vector, we'd never delete this redemption. With the buffer, we place a limit on how many outstanding redemptions can be held in memory

Reply:
Switched to `std::list` in https://github.com/Chatterino/chatterino2/pull/4949/commits/501022376af5bfb0210fc203b7e858b26774308a

In reality, most of the operations are (1) add element to empty collection and (2) remove singular element from collection (rendering it empty). This is $O(1)$ with `std::list` but we may want to consider whether `std::vector` has a small memory footprint here (if we prefer to optimize for the typical use case rather than edge cases)

Reply:
actually buffer may be best (i overestimated copying cost): https://www.codeproject.com/Articles/1185449/Performance-of-a-Circular-Buffer-vs-Vector-Deque-a / https://www.codeproject.com/Articles/340797/Number-crunching-Why-you-should-never-ever-EVER-us